src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,31 +6,129 @@
 
 package edu.ie3.osmogrid.io.input
 
-import akka.actor.typed.Behavior
-import akka.actor.typed.ActorRef
 import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
 import edu.ie3.osmogrid.guardian.OsmoGridGuardian.OsmoGridGuardianEvent
+import edu.ie3.osmogrid.io.input.InputDataProvider.readPbf
+import edu.ie3.util.osm.OsmEntities.{Node, OpenWay, Relation, Way}
+import edu.ie3.util.osm.OsmModel
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
+
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
 
 object InputDataProvider {
 
   sealed trait InputDataEvent
-  final case class Read()
-      extends InputDataEvent // todo this read method should contain configuration parameters for the actual source + potential filter options
+  final case class ReqOsm(replyTo: ActorRef[Response]) extends InputDataEvent
+  final case class ReqAssetTypes(replyTo: ActorRef[Response])
+      extends InputDataEvent
+
+  sealed trait Response
+  final case class RepOsm(osmModel: OsmModel) extends Response
+  final case class RepAssetTypes(osmModel: OsmModel) extends Response
+
   final case class Terminate(replyTo: ActorRef[OsmoGridGuardianEvent])
       extends InputDataEvent
 
-  def apply(cfg: OsmoGridConfig.Input): Behavior[InputDataEvent] =
-    Behaviors.receive { (ctx, msg) =>
-      msg match {
-        case _: Read =>
-          ctx.log.warn("Reading of data not yet implemented.")
-          Behaviors.same
-        case Terminate(_) =>
-          ctx.log.info("Stopping input data provider")
-          // TODO: Any closing of sources and stuff
-          Behaviors.stopped
-      }
+  def apply(cfgInput: OsmoGridConfig.Input): Behavior[InputDataEvent] =
+    Behaviors.receive[InputDataEvent] {
+      case (_, ReqOsm(replyTo)) =>
+        replyTo ! RepOsm(readPbf(cfgInput.osm.pbf.get.file))
+        Behaviors.same
```

Comment:
```suggestion
        readPbf(cfgInput.osm.pbf.get.file) match {
          case Success(osmModel) =>
            replyTo ! RepOsm(osmModel)
            Behaviors.same
          case Failure(exception) =>
            ctx.log.error(s"Unable to read osm information from file '${cfgInput.osm.pbf.get.file}'.")
            Behaviors.stopped
        }
      case (ctx, ReqAssetTypes(_)) =>
        ctx.log.info("Got request to provide asset types. But do nothing.")
        Behaviors.same
```

That's the way, you would handle errors in reading the file in Scala. Pretty nice, or not?

Additionally, you might wrap around another pattern matching for `cfgInput.osm.pbf` to avoid directly calling `get` on the Optional value. You can use that then for error logging as well.